packages/excalidraw/components/App.tsx

```diff
@@ -3524,18 +3563,57 @@ class App extends React.Component<AppProps, AppState> {
       elements?: SceneData["elements"];
       appState?: Pick<AppState, K> | null;
       collaborators?: SceneData["collaborators"];
-      commitToHistory?: SceneData["commitToHistory"];
+      commitToStore?: SceneData["commitToStore"];
     }) => {
-      if (sceneData.commitToHistory) {
-        this.history.resumeRecording();
+      const nextElements = restoreFractionalIndices(sceneData.elements ?? []);
+
+      if (sceneData.commitToStore) {
+        this.store.shouldCaptureIncrement();
+      }
+
+      if (sceneData.elements || sceneData.appState) {
+        let nextCommittedAppState = this.state;
+        let nextCommittedElements: Map<string, ExcalidrawElement>;
+
+        if (sceneData.appState) {
+          nextCommittedAppState = {
+            ...this.state,
+            ...sceneData.appState, // Here we expect just partial appState
+          };
+        }
+
+        const prevElements = this.scene.getElementsIncludingDeleted();
+
+        if (sceneData.elements) {
+          /**
+           * We need to schedule a snapshot update, as in case `commitToStore` is false  (i.e. remote update),
+           * as it's essential for computing local changes after the async action is completed (i.e. not to include remote changes in the diff).
+           *
+           * This is also a breaking change for all local `updateScene` calls without set `commitToStore` to true,
+           * as it makes such updates impossible to undo (previously they were undone coincidentally with the switch to the whole previously captured snapshot by history).
+           *
+           * WARN: be careful here as moving it elsewhere coulb break the history for remote client without noticing
+           * - we need to find a way to test two concurrent client updates simultaneously, while having access to both stores & histories.
+           */
+          this.store.shouldUpdateSnapshot();
+
+          nextCommittedElements = this.store.ignoreUncomittedElements(
```

Comment:
As written in the description of `ignoreUncomittedElements`, once we will be exchanging just increments and ephemerals (i.e. yet uncomitted actions) won't be recorded, this filtering won't be necessary.